web3/_utils/module_testing/personal_module.py

```diff
@@ -169,3 +230,117 @@ def test_personal_sign_and_ecrecover(self,
         )
         signer = web3.parity.personal.ecRecover(message, signature)
         assert is_same_address(signer, unlockable_account_dual_type)
+
+    def test_personal_sign_typed_data(self,
+                                      web3,
+                                      unlockable_account_dual_type,
+                                      unlockable_account_pw):
+        typed_message = '''
+            {
+                "types": {
+                    "EIP712Domain": [
+                        {"name": "name", "type": "string"},
+                        {"name": "version", "type": "string"},
+                        {"name": "chainId", "type": "uint256"},
+                        {"name": "verifyingContract", "type": "address"}
+                    ],
+                    "Person": [
+                        {"name": "name", "type": "string"},
+                        {"name": "wallet", "type": "address"}
+                    ],
+                    "Mail": [
+                        {"name": "from", "type": "Person"},
+                        {"name": "to", "type": "Person"},
+                        {"name": "contents", "type": "string"}
+                    ]
+                },
+                "primaryType": "Mail",
+                "domain": {
+                    "name": "Ether Mail",
+                    "version": "1",
+                    "chainId": "0x01",
+                    "verifyingContract": "0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC"
+                },
+                "message": {
+                    "from": {
+                        "name": "Cow",
+                        "wallet": "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826"
+                    },
+                    "to": {
+                        "name": "Bob",
+                        "wallet": "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB"
+                    },
+                    "contents": "Hello, Bob!"
+                }
+            }
+        '''
+        signature = HexBytes(web3.parity.personal.signTypedData(
+            json.loads(typed_message),
+            HexBytes(unlockable_account_dual_type).hex(),
+            unlockable_account_pw
+        ))
+
+        expected_signature = HexBytes(
+            "0xc8b56aaeefd10ab4005c2455daf28d9082af661ac347cd"
+            "b612d5b5e11f339f2055be831bf57a6e6cb5f6d93448fa35"
+            "c1bd56fe1d745ffa101e74697108668c401c"
+        )
+        assert signature == expected_signature
+        assert len(signature) == 32 + 32 + 1
+
+    def test_invalid_personal_sign_typed_data(self,
+                                              web3,
+                                              unlockable_account_dual_type,
+                                              unlockable_account_pw):
+        invalid_typed_message = '''
```

Comment:
This test looks equivalent to the one above and I don't see anywhere that it asserts the signature being invalid.. am I missing something?

Reply:
@pipermerriam this was incomplete when I pushed it. It has now been modified so that an invalid test has been produced and made sure that an error is raised.